Sources/TuistGraph/Models/Target.swift

```diff
@@ -5,7 +5,7 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
     // MARK: - Static
 
     public static let validSourceExtensions: [String] = [
-        "m", "swift", "mm", "cpp", "cc", "c", "d", "s", "intentdefinition", "xcmappingmodel", "metal", "mlmodel",
+        "m", "swift", "mm", "cpp", "cc", "c", "d", "s", "intentdefinition", "xcmappingmodel", "metal", "mlmodel", "docc",
```

Comment:
shouldn't this be in `validFolderExtensions`, as for `xcassets`?

Reply:
Great question! I had dove down that route originally and discovered that adding the `.docc` archive as a folder doesn’t allow Tuist to pick it up during generation how we’d like it to since `xcasset` and `.docc` are handled differently! I made a mini note of it in the PR description as well! 🙂 

If you create a new project in Xcode that simply contains a framework and it includes docc documentation, you’ll see the following in the build phases for the project.

<img width="1448" alt="Screen Shot 2022-01-13 at 12 48 14 PM" src="https://user-images.githubusercontent.com/26095410/149385183-4e7c0d53-23e4-4128-b0f4-d2a79df64f2e.png">

This part was weird for me to wrap my head around, the documentation archive is treated as though it’s a compile source rather than a resource like I’d expect. Adding it as a folder won’t allow this type of behavior. I’d be open to suggestions on how to better approach this one! :) I didn’t like adding a “folder” to a “validSourceExtension” variable either! 🤮